src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -3650,6 +3650,20 @@ static QVariant fcnStraightDistance2d( const QVariantList &values, const QgsExpr
   return QVariant( curve->straightDistance2d() );
 }
 
+static QVariant fcnRoundness( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QgsGeometry geom = QgsExpressionUtils::getGeometry( values.at( 0 ), parent );
+  const QgsCurvePolygon *poly = geom.constGet() ? qgsgeometry_cast< const QgsCurvePolygon * >( geom.constGet()->simplifiedTypeRef() ) : nullptr;
+
+  if ( !poly )
+  {
+    parent->setEvalErrorString( QObject::tr( "Function `roundness` requires a polygon geometry or a multi polygon geometry with a single part." ) );
```

Comment:
Can you add tests for multip olygon too?

Reply:
There are already tests for multi polygons with single part and multi parts, no ?

Reply:
Only for the expression, right? You can at least add it in `tests/src/core/geometry/testqgscurvepolygon.cpp`

Reply:
Sorry for my comment... I misread the description.